quality/views.py

```
@@ -0,0 +1,2496 @@
+    from .models import IntelligentQualityReport
+
+    try:
+        report = IntelligentQualityReport.objects.select_related(
+            'production_line', 'generated_by'
+        ).get(id=report_id)
+
+        # Verificar permissão
+        if not (request.user.is_superuser or
+                request.user.groups.filter(name__icontains='qualidade').exists() or
+                report.generated_by == request.user):
+            return HttpResponseForbidden("Você não tem permissão para visualizar este relatório.")
```

Ruff (0.11.9)

Undefined name `HttpResponseForbidden`

(F821)